experimental/dds/tree2/src/core/forest/forest.ts

```diff
@@ -72,6 +72,11 @@ export interface IForestSubscription extends Dependee, ISubscribable<ForestEvent
 	 */
 	forgetAnchor(anchor: Anchor): void;
 
+	/**
+	 * Retrieves all detached fields on the forest, including the root field.
+	 */
+	getDetachedFields(): DetachedField[];
```

Comment:
After thinking about this a bit more, I think we might want to go with a different approach here.

Doing it this way means that the summary has to encode the different fields separately, and when we move to chunked forest's compressed format, it will work a lot better if we encode it all at once. Thus we could instead provide a way to get a cursor over the root fields. I believe both our forests have a dummy node above the detached fields thats supposed to be an implementation detail which we can point a cursor at to implement this. Infact, both forests have a private "moveCursorToPath" that allows this. We could simple make those public and add it to the forest interface. PAssing undefined as the destination to that function will navigate to this above root dummy node. The docs for the function will need to be carefully worded about that, but I think it will be ok, especially if we don't include forest in our package API.

Reply:
I like this approach a lot more, thanks!